doc/cprover-manual/loop-invariant-synthesizer.md

```diff
@@ -0,0 +1,37 @@
+[CPROVER Manual TOC](../)
+
+## Loop Invariant Synthesis with goto-instrument
+
+This is a loop invariant synthesizer built in goto-instrument that 
+synthesize [loop invariants](contracts-invariants.md) for a 
+pre-existing GOTO binary, annotate the invariant clause into 
+the GOTO binary, and produce an annotated binary as output.
+
+All synthesized loop invariants must be inductive, i.e., 
+all [loop-invariant checks](contracts-invariants.md#semantics)
+succeed in the instrumented GOTO binary.
+
+For different synthesis strategies, there are different guarantees 
+beyond the inductiveness on the synthesized loop invariants, 
+summarized in the following table.
+
+Strategy         | Flag                                           | Guarantees
+-----------------|------------------------------------------------|----
+Enumerative      | --synthesize-loop-invariants-enumerative       | All checks will succeeded in the annotated GOTO binary
+
+### Enumerative Synthesizer
+
+This synthesizer is designed for GOTO binary that contains only checks instrumented 
+with the flag  ```--pointer-check```, and synthesizes loop invariants that make
+all checks succeeded. The idea of the synthesizer is first inferring a clause, 
+using the idea of weakest preconditions, for each violated check to make it succeed, 
+and then enumerating a strengthening clause to make the inferred clause inductive.
```

Comment:
```suggestion
The enumerative synthesizer is designed for GOTO binaries instrumented only for pointer checks
(i.e., using ```--pointer-check``` flag), because it synthesizes loop invariants that make
all pointer checks succeeded. First, the synthesizer infer a clause for each violated check
(based on weakest preconditions) in order to make them succeed. Then, it enumerates
a strengthening clause to make it inductive.
```